wremnants/recoil_tools.py

```diff
@@ -686,23 +689,22 @@ def add_recoil_unc_Z(self, df, results, dataset, cols, axes, hName):
     def add_recoil_unc_W(self, df, results, dataset, cols, axes, hName):
         return self.add_recoil_unc_Z(df, results, dataset, cols, axes, hName) # currently use the Z implementation
 
-    def setup_recoil_Z_unc(self):
-        if not self.dataset.name in self.datasets_to_apply or not self.storeHists:
-            return
+    #def setup_recoil_unc_common(self, hNames, cols, axes):
 
+    def setup_recoil_Z_unc(self):
         hNames, cols, axes = [], [], [] 
         if self.storeHists:
             hNames = ["recoil_corr_rec_para_qt", "recoil_corr_rec_para", "recoil_corr_rec_perp", "recoil_corr_rec_magn", "met_corr_rec_pt", "mt_corr_rec"]
             cols = hNames
             axes = [self.axis_recoil_para_qT, self.axis_recoil_para, self.axis_recoil_perp, self.axis_recoil_magn, self.axis_MET_pt, self.axis_mt]
 
+        if not self.dataset.name in self.datasets_to_apply:
```

Comment:
why moving it down, should it have an effect?

Reply:
This is an artefact of trying to debug some other stuff, will fix it